distribution.pm

```diff
@@ -104,6 +104,89 @@ sub become_root {
 }
 
+sub connection_hijack {
+    my ($self) = @_;
+    return unless testapi::get_var('CONNECTIONS_HIJACK_PROXY') || testapi::get_var('CONNECTIONS_HIJACK_DNS') || testapi::get_var('CONNECTIONS_HIJACK');
+
+    my $dns_port   = testapi::get_var('CONNECTIONS_HIJACK_DNS_SERVER_PORT');
+    my $hostname   = testapi::get_var('WORKER_HOSTNAME') || '10.0.2.2';
+    my $proxy_port = testapi::get_var('CONNECTIONS_HIJACK_PROXY_SERVER_PORT');
+    my $fakeip     = testapi::get_var('CONNECTIONS_HIJACK_FAKEIP');
+
+    bmwqemu::fctwarn(
+"You have specified connection hijack, but you are not in qemu user mode. This means that you should setup guest tcp redirection manually to the host for http requests"
+    ) unless testapi::get_var('NICTYPE') eq "user";
+
+    if (   testapi::get_var('CONNECTIONS_HIJACK_PROXY_ENTRY')
+        && !testapi::get_var('CONNECTIONS_HIJACK_DNS')
+        && testapi::get_var('CONNECTIONS_HIJACK_FAKEIP')
+        && testapi::get_var('NICTYPE') eq "user")
+    {
+        my $hosts = testapi::get_var('CONNECTIONS_HIJACK_PROXY_ENTRY');
+
+        my @entry = split(/,/, $hosts);
+        return unless (@entry > 0);
+
+        # Generate record table from configuration
+        my @redirect_hosts = map {
+            my ($host, $redirect) = split(/:/, $_);
+            $host if ($host)
+        } @entry;
+
+        $self->set_host_entry($fakeip, $_) for @redirect_hosts;
+
+    }
+    elsif (!testapi::get_var('LIVECD')) {    #LiveCD environments sadly does not support iptables calls.
+        if (testapi::get_var('CONNECTIONS_HIJACK') || testapi::get_var('CONNECTIONS_HIJACK_DNS')) {
+            testapi::script_run("iptables -t nat -A OUTPUT -p udp --dport 53 -j DNAT --to-destination $hostname:$dns_port");
+        }
+
+        if (testapi::get_var('CONNECTIONS_HIJACK') || testapi::get_var('CONNECTIONS_HIJACK_PROXY')) {
+            testapi::script_run("iptables -t nat -A OUTPUT -p tcp --dport 80 -j DNAT --to-destination $hostname:$proxy_port");
+            testapi::script_run("iptables -t nat -A OUTPUT -p tcp --dport 443 -j DNAT --to-destination $hostname:$proxy_port");
+        }
+    }
+    else {
+
+        my $os   = "linux";
+        my $arch = "amd64";
+
+        $arch = "386"   if (testapi::check_var('ARCH', 'i586'));
+        $arch = "arm64" if (testapi::check_var('ARCH', 'aarch64'));
+        $arch = "arm"   if (testapi::check_var('ARCH', 'armhfp') || testapi::check_var('ARCH', 'arm') || testapi::check_var('ARCH', 'armv7'));
+
+        $os = "windows" if (testapi::check_var('DISTRI', 'windows'));
+        $os = "darwin"  if (testapi::check_var('DISTRI', 'macosx'));
+        $os = "openbsd" if (testapi::check_var('DISTRI', 'openbsd'));
+        $os = "freebsd" if (testapi::check_var('DISTRI', 'freebsd'));
+
+
+        # Unfortunately as a fallback, we need an agent that proxies UDP requests from the guest to the host machine
+        # since UDP guestfwd it's not supported, nor iptables is present on LIVECDs
```

Comment:
Didn't tested installation yet, but i'm pretty sure will have the same problem.  i'll let you know when as soon as i setup a test for this.